src/ImageSharp/Formats/Ico/IcoFrameMetadata.cs

```diff
@@ -0,0 +1,119 @@
+    /// </summary>
+    public IcoFrameMetadata()
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="IcoFrameMetadata"/> class.
+    /// </summary>
+    /// <param name="width">width</param>
+    /// <param name="height">height</param>
+    /// <param name="colorCount">colorCount</param>
+    /// <param name="planes">planes</param>
+    /// <param name="bitCount">bitCount</param>
+    /// <param name="bytesInRes">bytesInRes</param>
+    /// <param name="imageOffset">imageOffset</param>
+    public IcoFrameMetadata(byte width, byte height, byte colorCount, ushort planes, ushort bitCount, uint bytesInRes, uint imageOffset)
+    {
+        this.Width = width;
+        this.Height = height;
+        this.ColorCount = colorCount;
+        this.Planes = planes;
+        this.BitCount = bitCount;
+        this.BytesInRes = bytesInRes;
+        this.ImageOffset = imageOffset;
+    }
+
+    private IcoFrameMetadata(IcoFrameMetadata metadata)
+    {
+        this.Width = metadata.Width;
+        this.Height = metadata.Height;
+        this.ColorCount = metadata.ColorCount;
+        this.Planes = metadata.Planes;
+        this.BitCount = metadata.BitCount;
+        this.BytesInRes = metadata.BytesInRes;
+        this.ImageOffset = metadata.ImageOffset;
+        this.BmpMetadata = metadata.BmpMetadata;
+        this.PngMetadata = metadata.PngMetadata;
+    }
+
+    /// <summary>
+    /// Gets or sets BmpMetadata.
+    /// </summary>
+    public Bmp.BmpMetadata? BmpMetadata { get; set; }
+
+    /// <summary>
+    /// Gets or sets PngMetadata.
+    /// </summary>
+    public Png.PngMetadata? PngMetadata { get; set; }
+
+    /// <summary>
+    /// Gets or sets Width field. <br />
+    /// Specifies image width in pixels. Can be any number between 0 and 255. Value 0 means image width is 256 pixels.
+    /// </summary>
+    public byte Width { get; set; }
+
+    /// <summary>
+    /// Gets or sets Height field. <br />
+    /// Specifies image height in pixels. Can be any number between 0 and 255. Value 0 means image height is 256 pixels.
+    /// </summary>
+    public byte Height { get; set; }
+
+    /// <summary>
+    /// Gets or sets ColorCount field. <br />
+    /// Specifies number of colors in the color palette. Should be 0 if the image does not use a color palette.
+    /// </summary>
+    public byte ColorCount { get; set; }
+
+    /// <summary>
+    /// Gets or sets Planes field. <br />
+    /// In ICO format: Specifies color planes. Should be 0 or 1. <br />
+    /// In CUR format: Specifies the horizontal coordinates of the hotspot in number of pixels from the left.
+    /// </summary>
+    public ushort Planes { get; set; }
+
+    /// <summary>
+    /// Gets or sets BitCount field. <br />
+    /// In ICO format: Specifies bits per pixel. <br />
+    /// In CUR format: Specifies the vertical coordinates of the hotspot in number of pixels from the top.
+    /// </summary>
+    public ushort BitCount { get; set; }
```

Comment:
Ambiguity like this makes me want to split the formats out into separate ones with shared utility methods.